.ci/maven-release-perform.sh

```diff
@@ -26,7 +26,7 @@ SKIP_OTHERS="-Dpmd.skip=true -Dspotbugs.skip=true -Djacoco.skip=true -Dxml.skip=
 
 git checkout "checkstyle-$TARGET_VERSION"
 echo "Deploying jars to maven central (release:perform) ..."
-mvn -e --no-transfer-progress -Pgpg release:perform \
+mvn -e --no-transfer-progress -B release:perform \
```

Comment:
Batch mode `-B` needs to be used since download progress was present and was clogging the logs.

Reply:
Also, `-Pgpg` mode should be excluded because `setup-java` takes care of that.

Reply:
https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#publishing-using-apache-maven

The generated `$HOME/.m2/settings.xml` by the action would be:
```xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <servers>
    <server>
      <id>maven</id>
      <username>${env.MAVEN_USERNAME}</username>
      <password>${env.MAVEN_CENTRAL_TOKEN}</password>
    </server>
    <server>
      <id>gpg.passphrase</id>
      <passphrase>${env.MAVEN_GPG_PASSPHRASE}</passphrase>
    </server>
  </servers>
</settings>
```

Extra setup that I did as well: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#extra-setup-for-pomxml

The action takes care of importing the gpg key into the keychain.

The `GPG_KEY` secret should be exported as `gpg --armor --export-secret-keys YOUR_ID`: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#gpg

Proof is the action itself `maven-gpg-plugin`: https://github.com/stoyanK7/checkstyle/actions/runs/4183171423/jobs/7247190315#step:7:3731

Reply:
Returned back to `-Dgpg`. Done.

Reply:
@stoyanK7 , why we need `-B` ? if we have problems with too much logs , ` --no-transfer-progress` should care about this.

Reply:
https://stackoverflow.com/a/48929096/9553927
> Batch mode causes Maven to not display "Progress: 125/150kB" style lines when running. If you are running Maven on some server and then checking the logs afterwards, these progress lines take up 90% of the log and make it basically impossible to find the stuff that matters. Setting batch mode prevents this. Apart from that, I don't know any other use for batch mode. As others have said, I have never seen Maven prompt for anything during a build, regardless of whether interactive or batch mode is set.

---
https://github.com/checkstyle/checkstyle/pull/12733#issuecomment-1428406622

> Operation was cancelled so there is no reasoning. There is an excessive amount of downloads in the logs. Please ensure maven commands are turning off downloads, otherwise logs become hard to read. Also make sure you run in Batch mode to ensure it is not trying to get input from a "user".

---

> > 
> 
>     excessive amount of downloads in the logs
>     Batch mode
> 
> Looks like update needs to be made to 

https://github.com/checkstyle/checkstyle/pull/12733#issuecomment-1428425571

Reply:
For some reason `--no-transfer-progress` isn't getting picked up. We also use batch mode in mvn:preprare.

Reply:
Here are logs without `-B`: https://github.com/stoyanK7/checkstyle/actions/runs/4225686733/jobs/7338275777#step:6:16641

Reply:
@romani 
Done. Removed.

Uh, just found out that `--no-transfer-progess` needs to be added as an argument after `release:perform` to work. I was wondering why it was not working and hence added `-B`

Here is the latest run with the latest changes:
https://github.com/stoyanK7/checkstyle/actions/runs/4225875334/jobs/7338675470

Reply:
Strange , such argument is for mvn , not any other plugins, but it is ok to keep it as it works